agent/agent_pool.go

```diff
@@ -53,16 +53,18 @@ func (r *AgentPool) runWorker(worker *AgentWorker, im *IdleMonitor) error {
 	}
 
 	// Starts the agent worker and wait for it to finish
-	if err := worker.Start(im); err != nil {
-		return err
-	}
+	err := worker.Start(im)
```

Comment:
Perhaps a comment here ala "we need to ensure the agent disconnects successfully if a worker process fails to start, so this error will be checked later on"

Reply:
I also thought about using a `defer` after `Connect`, would that be more obvious without a comment?

Reply:
Oh yeah, good idea.